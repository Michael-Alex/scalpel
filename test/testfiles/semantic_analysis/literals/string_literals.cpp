char str1[5] = "test";
char str2[6] = " A \n ";
char str3[3] = "\u0060b";
char str4[4] = "\U000000ffz";

wchar_t wstr1[5] = L"test";
wchar_t wstr2[6] = L" A \n ";
wchar_t wstr3[3] = L"\u0060b";
wchar_t wstr4[3] = L"\U000000ffz";

