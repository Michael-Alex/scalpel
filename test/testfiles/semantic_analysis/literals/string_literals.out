<namespace id="0">
 <variable id="0" name="str1">
  <type>
   <array size="5">
    <fundamental_type type="char"/>
   </array>
  </type>
  <default_value>
   <expression type="string">
    74 65 73 74 
   </expression>
  </default_value>
 </variable>
 <variable id="1" name="str2">
  <type>
   <array size="6">
    <fundamental_type type="char"/>
   </array>
  </type>
  <default_value>
   <expression type="string">
    20 41 20 a 20 
   </expression>
  </default_value>
 </variable>
 <variable id="2" name="str3">
  <type>
   <array size="3">
    <fundamental_type type="char"/>
   </array>
  </type>
  <default_value>
   <expression type="string">
    60 62 
   </expression>
  </default_value>
 </variable>
 <variable id="3" name="str4">
  <type>
   <array size="4">
    <fundamental_type type="char"/>
   </array>
  </type>
  <default_value>
   <expression type="string">
    c3 bf 7a 
   </expression>
  </default_value>
 </variable>
 <variable id="4" name="wstr1">
  <type>
   <array size="5">
    <fundamental_type type="wchar_t"/>
   </array>
  </type>
  <default_value>
   <expression type="wide string">
    74 65 73 74 
   </expression>
  </default_value>
 </variable>
 <variable id="5" name="wstr2">
  <type>
   <array size="6">
    <fundamental_type type="wchar_t"/>
   </array>
  </type>
  <default_value>
   <expression type="wide string">
    20 41 20 a 20 
   </expression>
  </default_value>
 </variable>
 <variable id="6" name="wstr3">
  <type>
   <array size="3">
    <fundamental_type type="wchar_t"/>
   </array>
  </type>
  <default_value>
   <expression type="wide string">
    60 62 
   </expression>
  </default_value>
 </variable>
 <variable id="7" name="wstr4">
  <type>
   <array size="3">
    <fundamental_type type="wchar_t"/>
   </array>
  </type>
  <default_value>
   <expression type="wide string">
    ff 7a 
   </expression>
  </default_value>
 </variable>
</namespace>
